### PR TITLE
Single end adapter

### PR DESCRIPTION
**Pull request by msettles**
Address #87 and #115, first support for adapter removal in SE reads. Can maybe use some parameter optimization but works surprisingly well. Used the same core as overlap, where I overlap adapter with SE read, if overlap found, the trim off starting at overlap. Uses same parameters as overlap, quite NICE!

### CODE REVIEW

File: hts_AdapterTrimmer/src/hts_AdapterTrimmer.cpp [hunk 1]
```diff
@@ -61,6 +61,8 @@ int main(int argc, char** argv)
 
         desc.add_options()
             ("no-fixbases,X", po::bool_switch()->default_value(false), "after trimming adapter, DO NOT use consensus sequence of paired reads, only trims adapter sequence");
+        desc.add_options()
+            ("adapter-sequence,a", po::value<std::string>()->default_value("AGATCGGAAGAGCACACGTCTGAACTCCAGTCA"),  "Primer sequence to trim in SE adapter trimming, default is truseq ht primer sequence");
```
Comment (dstreett) [inline hts_AdapterTrimmer/src/hts_AdapterTrimmer.cpp]: Just checking - do we want fasta input here?

**Comment by dstreett** [comment]
Looks good!

A quick questions. How do we want this to handle if you have 3 bases pairs at the end that seem to be adapters (or any K-1 length adapter sequence)? 

With the tests you put up I'm fine accepting this merge request as is right now.

File: hts_AdapterTrimmer/src/hts_AdapterTrimmer.cpp (hunk 1 above)
Comment (msettles) [reply hts_AdapterTrimmer/src/hts_AdapterTrimmer.cpp]: Maybe later, for multiple fastqs, will add as enhancement to issues

**Comment by msettles** [comment]
We'll need to determine how for to go on the end of reads, some testing is in order

**Pull request merged**